src/main/java/com/zenuml/confluence/macro/SequenceMacro.java

```diff
@@ -2,28 +2,76 @@
 
 import com.atlassian.confluence.content.render.xhtml.ConversionContext;
 import com.atlassian.confluence.macro.*;
+import com.atlassian.confluence.pages.Attachment;
+import com.atlassian.confluence.pages.AttachmentManager;
+import com.atlassian.confluence.setup.settings.SettingsManager;
 import com.atlassian.plugin.spring.scanner.annotation.component.Scanned;
 import com.atlassian.plugin.spring.scanner.annotation.imports.ComponentImport;
 import com.atlassian.webresource.api.assembler.PageBuilderService;
 import org.springframework.beans.factory.annotation.Autowired;
 
+import javax.xml.bind.DatatypeConverter;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
 import java.util.Map;
 
 @Scanned
 public class SequenceMacro implements Macro {
 
     private PageBuilderService pageBuilderService;
+    private AttachmentManager attachmentManager;
+    private SettingsManager settingsManager;
 
     @Autowired
-    public SequenceMacro(@ComponentImport PageBuilderService pageBuilderService) {
+    public SequenceMacro(
+            @ComponentImport PageBuilderService pageBuilderService,
+            @ComponentImport AttachmentManager attachmentManager,
+            @ComponentImport SettingsManager settingsManager
+    ) {
         this.pageBuilderService = pageBuilderService;
+        this.attachmentManager = attachmentManager;
+        this.settingsManager = settingsManager;
     }
 
+    public String createMD5(String plaintext) {
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            md.update(plaintext.getBytes());
+            byte[] digest = md.digest();
+            String myHash = DatatypeConverter.printHexBinary(digest).toLowerCase();
+            return  myHash;
+        } catch (NoSuchAlgorithmException e) {
+            return null;
```

Comment:
This one I agree, when I was doing this I was straggling with throw error and return null.
After I throw an error here the UML will be shown as a red error block, so I was not sure whether it is good or not. The purpose to return null here is to show the dom rather than img tag if something goes wrong.